src/linear_algebra/clifford_algebra/contraction.lean

```diff
@@ -0,0 +1,313 @@
+by simpa only [map_one] using apply_dual_left_algebra_map Q d 1
+
+@[simp] lemma apply_dual_right_one : (1 : clifford_algebra Q) ⌊ d = 0 :=
+by simpa only [map_one] using apply_dual_right_algebra_map Q d 1
+
+variables {Q}
+
+/-- This is [grinberg_clifford_2016][] Theorem 7 -/
+lemma apply_dual_left_apply_dual_left (x : clifford_algebra Q) :
+  d ⌋ (d ⌋ x) = 0 :=
+begin
+  induction x using clifford_algebra.left_induction with r x y hx hy m x hx,
+  { simp_rw [apply_dual_left_algebra_map, map_zero] },
+  { rw [map_add, map_add, hx, hy, add_zero] },
+  { rw [apply_dual_left_ι_mul, map_sub, apply_dual_left_ι_mul, hx, linear_map.map_smul, mul_zero,
+      sub_zero, sub_self], }
+end
+
+/-- This is [grinberg_clifford_2016][] Theorem 13 -/
+lemma apply_dual_right_apply_dual_right (x : clifford_algebra Q) :
+  (x ⌊ d) ⌊ d = 0 :=
+by rw [apply_dual_right_eq, apply_dual_right_eq, reverse_reverse, apply_dual_left_apply_dual_left,
+  map_zero]
+
+/-- This is [grinberg_clifford_2016][] Theorem 8 -/
+lemma apply_dual_left_comm (x : clifford_algebra Q) : d ⌋ (d' ⌋ x) = -(d' ⌋ (d ⌋ x)) :=
+begin
+  induction x using clifford_algebra.left_induction with r x y hx hy m x hx,
+  { simp_rw [apply_dual_left_algebra_map, map_zero, neg_zero] },
+  { rw [map_add, map_add, map_add, map_add, hx, hy, neg_add] },
+  { simp only [apply_dual_left_ι_mul, map_sub, linear_map.map_smul],
+    rw [neg_sub, sub_sub_eq_add_sub, hx, mul_neg, ←sub_eq_add_neg] }
+end
+
+/-- This is [grinberg_clifford_2016][] Theorem 14 -/
+lemma apply_dual_right_comm (x : clifford_algebra Q) : (x ⌊ d) ⌊ d' = -((x ⌊ d') ⌊ d) :=
+by rw [apply_dual_right_eq, apply_dual_right_eq, apply_dual_right_eq, apply_dual_right_eq,
+  reverse_reverse, reverse_reverse, apply_dual_left_comm, map_neg]
+
+/- TODO:
+lemma apply_dual_right_apply_dual_left (x : clifford_algebra Q) : (d ⌋ x) ⌊ d' = d ⌋ (x ⌊ d') :=
+-/
+
+end apply_dual_left
+
+local infix `⌋`:70 := apply_dual_left
+local infix `⌊`:70 := apply_dual_right
+
+/-- Auxiliary construction for `clifford_algebra.alpha` -/
+@[simps]
+def alpha_aux (B : bilin_form R M) : M →ₗ[R] clifford_algebra Q →ₗ[R] clifford_algebra Q :=
+begin
+  have v_mul := (algebra.lmul R (clifford_algebra Q)).to_linear_map ∘ₗ ι Q,
+  exact v_mul - (apply_dual_left ∘ₗ B.to_lin) ,
+end
+
+lemma alpha_aux_alpha_aux (B : bilin_form R M) (v : M) (x : clifford_algebra Q) :
+  alpha_aux Q B v (alpha_aux Q B v x) = (Q v - B v v) • x :=
+begin
+  simp only [alpha_aux_apply_apply],
+  rw [mul_sub, ←mul_assoc, ι_sq_scalar, map_sub, apply_dual_left_ι_mul, ←sub_add, sub_sub_sub_comm,
+    ←algebra.smul_def, bilin_form.to_lin_apply, sub_self, sub_zero, apply_dual_left_apply_dual_left,
+    add_zero, sub_smul],
+end
+
+variables {Q}
+
+variables {Q' Q'' : quadratic_form R M} {B B' : bilin_form R M}
+variables (h : B.to_quadratic_form = Q' - Q) (h' : B'.to_quadratic_form = Q'' - Q')
+
+/-- Convert between two algebras of different quadratic form -/
+def alpha (h : B.to_quadratic_form = Q' - Q) :
```

Comment:
Why this name?

Reply:
Only because it matches the name used in my reference. I was intending to try and see if bourbaki had a similar construction and to find out what names were used there.

I guess `replace_quaradic_form` or `change_form` or similar might be a more insightful name?

Reply:
Here's what Bourbaki has to say:

> PROPOSITION 3. - Soient $Q$ et $Q'$ deux formes quadratiques et $F$ une forme bilinéaire sur $E$ telles que $Q'(x) = Q(x) + F(x, x)$ pour tout $x \in E$. L'application $\lambda_F$, applique l'idéal $I(Q_1)$ sur l'idéal $I(Q)$, et définit un isomorphisme (noté $\overbar{\lambda_F}$) du $A$-module $C(Q')$ sur le $A$-module $C(Q)$.

Here, $\overbar{\lambda_F}$ is my `alpha_equiv hF`, and `\lambda_F` is my `alpha hF`.

So yes, `alpha` is clearly a name used by only my original reference, and not a good choice.

Reply:
I renamed it to `change_form` with a longer docstring and a bourbaki reference.